### ISSUE TITLE
Output format support

### ISSUE THREAD

**Issue by i-sinister** [DarkWanderer/ClickHouse.Client#36]
Quite often all appication is doing is just feeding query result to a "processing pipeline" in some format ("returning json from rest api to some client, maybe some SPA" is not an exception here). As clickhouse already [supports great variety of output formats](https://clickhouse.tech/docs/en/interfaces/formats/) that could be fed to the "pipeline" (json, protobuf, avro, arrow, parquet, csv) it looks reasonable to allow query results to be returned in desired format to cut ADO and formatting libraries all together. There can be an extension method on a reader object named, for example, "export", accepting stream where data should be copied:
```csharp
var command = connection.CreateCommand();
command.Text = ".... Format Protobuf";
using var reader = await command.ExecuteReaderAsync();
await reader.Export(output);
```
Another option is having extension method for command:
```csharp
var command = connection.CreateCommand();
command.Text = ".... Format Protobuf";
using var exporter = await command.ExecuteExportAsync();
await exporter.Export(output);
```
It is important to separate query execution from "copying results" to be able to handle errors

**Comment by DarkWanderer** [comment]
Hi,

Added `ExecuteRawResultAsync` function, example:
```
            using var command = connection.CreateCommand();
            command.CommandText = "SELECT 1 FORMAT TSV";
            using var result = await command.ExecuteRawResultAsync(CancellationToken.None);
            using var stream = await result.ReadAsStreamAsync();
            using var reader = new StreamReader(stream);
            Assert.AreEqual("1", reader.ReadToEnd().Trim());
```

**Comment by i-sinister** [comment]
Seems legit, ReadAsStreamAsync should accept CancellationToken though.

**Comment by DarkWanderer** [comment]
[Original method](https://docs.microsoft.com/ru-ru/dotnet/api/system.net.http.httpcontent.readasstreamasync) is not cancellable, so it's not possible

**Comment by i-sinister** [comment]
Actua [it is](https://github.com/dotnet/runtime/blob/master/src/libraries/System.Net.Http/src/System/Net/Http/HttpContent.cs#L260). I wonder why does msdn does not list the overload

**Comment by i-sinister** [comment]
My bad - there is overload in version 5, not 3

**Comment by DarkWanderer** [comment]
Released version 2.0.0 with this
